### PR TITLE
JuMP/MOI to new version v0.20/0.9

### PR DESCRIPTION
**Pull request by Wikunia**
Upgraded to JuMP v0.20 and MOI v0.9 (#14 )
Questions:
- There was a `TODO` to return the status of the optimization:
  I now throw an error such that user knows that this is not optimal. Is there value in a not optimal solution? Then we should return the status to the user but currently as long as this is not supported I think throwing an error is the safe option.
- Is it fast to call `is_bipartite` ? If not we might want to only check it if the solver is only a MIP solver? And maybe a function argument to specifically set that this graph is bipartite?

Additional deprecation changes:
- `CompleteGraph` -> `complete_graph`
- `CompleteBipartiteGraph` -> `complete_bipartite_graph`

Realized that this is missing:
- remove Julia 0.7
- change the Readme

Maybe @matbesancon wants to review this.

### CODE REVIEW

**Comment by matbesancon** [comment]
Could you remove travis support for 0.7 and add support for 1.0 and 1.2?

File: REQUIRE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 julia 0.7
 LightGraphs 1.2
-JuMP 0.18
-MathProgBase 0.7
+JuMP 0.20
+MathOptInterface 0.9.0
```
Comment (matbesancon) [inline REQUIRE]:
```suggestion
MathOptInterface 0.9
```

**Comment by matbesancon** [comment]
> There was a TODO to return the status of the optimization:
I now throw an error such that user knows that this is not optimal. Is there value in a not optimal solution? Then we should return the status to the user but currently as long as this is not supported I think throwing an error is the safe option.

What we could do is return a tuple with the optimization result and the solution, it becomes easier with MOI status enum

**Comment by matbesancon** [comment]
@Wikunia you can merge master into your branch, this will get you a working basis

**Comment by Wikunia** [comment]
> > There was a TODO to return the status of the optimization:
> > I now throw an error such that user knows that this is not optimal. Is there value in a not optimal solution? Then we should return the status to the user but currently as long as this is not supported I think throwing an error is the safe option.
> 
> What we could do is return a tuple with the optimization result and the solution, it becomes easier with MOI status enum

Saw this comment in `maximum_weight_matching.jl`
```
Returns MatchingResult containing:
  - a solve status (indicating whether the problem was solved to optimality)
  - the optimal cost
  - a list of each vertex's match (or -1 for unmatched vertices)
```

but I assume there was a reason to remove the status from the `MatchingResult` in general. In that case I can return `(status, MatchingResult)` in `maximum_weight_matching.jl` and `lp.jl`.

File: README.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ containing the `mate` and `weight` fields.
 ### Perfect matching
 
 ```julia
-g =CompleteGraph(4)
+g =complete_graph(4)
```
Comment (matbesancon) [inline README.md]:
```suggestion
g = complete_graph(4)
```

File: README.md [hunk 3]
```diff
@@ -16,7 +16,7 @@ containing the `mate` and `weight` fields.
...
 w =Dict{Edge,Float64}()
```
Comment (matbesancon) [inline README.md]:
```suggestion
w = Dict{Edge,Float64}()
```

File: src/LightGraphsMatching.jl [hunk 4]
```diff
@@ -5,7 +5,8 @@ using LightGraphs
 using SparseArrays: spzeros
 
 using JuMP
-using MathProgBase: AbstractMathProgSolver
+using MathOptInterface
+MOI = MathOptInterface
```
Comment (matbesancon) [inline src/LightGraphsMatching.jl]:
```suggestion
const MOI = MathOptInterface
```

File: src/maximum_weight_matching.jl [hunk 5]
```diff
@@ -49,15 +49,16 @@ function maximum_weight_matching(g::Graph,
     @objective(model, Max, sum(x[e]*w[src(e),dst(e)] for e in edge_list))
 
     @constraint(model, c1[i=1:n], sum(x[Edge(minmax(i,j))] for j in neighbors(g,i)) <= 1)
-    status = solve(model)
-    solution = getvalue(x)
-    cost = getobjectivevalue(model)
-    ## TODO: add the option of returning the solve status as part of the MatchingResult type.
+    optimize!(model)
+    status = JuMP.termination_status(model)
+    status != MOI.OPTIMAL  && error("JuMP solver failed to find optimal solution.")
```
Comment (matbesancon) [inline src/maximum_weight_matching.jl]:
```suggestion
    status != MOI.OPTIMAL && error("JuMP solver failed to find optimal solution.")
```

**Comment by matbesancon** [comment]
Thanks!

**Pull request merged**